app/models/document.rb

```diff
@@ -103,13 +107,81 @@ def public_updated_at
   end
 
   def public_updated_at=(timestamp)
-    @public_updated_at = Time.parse(timestamp) unless timestamp.nil?
+    @public_updated_at = Time.parse(timestamp.to_s) unless timestamp.nil?
   end
 
-private
+  # ACTIVE RECORD LIKE INTERFACE
```

Comment:
I would either delete this comment or if it does feel like a separate concern, think about pulling it into a mixin.